arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/option.kt

```diff
@@ -22,8 +22,11 @@ public value class OptionEffectScope(private val cont: EffectScope<None>) : Effe
   public suspend fun <B> Option<B>.bind(): B =
     bind { None }
 
-  public suspend fun <B> B?.bind(): B =
-    this ?: shift(None)
+  @OptIn(ExperimentalContracts::class)
+  public suspend fun <B> B?.bind(): B {
+    contract { returns() implies (this@bind != null) }
```

Comment:
This doesn't conflict with `Either<None, A>.bind()` inherited from `EffectScope<None>?

Reply:
In essence, this is the equivalent as `ensureNotNull` so it's kinda redundant.

Reply:
Agree